src/jdk.jlink/share/classes/jdk/tools/jlink/internal/JlinkTask.java

```diff
@@ -802,6 +828,64 @@ private static class ImageHelper implements ImageProvider {
                                 .collect(Collectors.toSet());
         }
 
+        private static Platform targetPlatform(Configuration cf, Map<String, Path> modsPaths) throws IOException {
+            Path javaBasePath = modsPaths.get("java.base");
+            assert javaBasePath != null : "java.base module path is missing";
+            if (isJavaBaseFromCurrentPlatform(javaBasePath)) {
+                // this implies that the java.base module used for the target image
+                // will correspond to the current platform. So this isn't an attempt to
+                // build a cross-platform image. We use the current platform's endianness
+                // in this case
+                return Platform.runtime();
+            } else {
+                // this is an attempt to build a cross-platform image. We now attempt to
+                // find the target platform's arch and thus its endianness from the java.base
+                // module's ModuleTarget attribute
+                String targetPlatformVal = readJavaBaseTargetPlatform(cf);
+                try {
+                    return Platform.parsePlatform(targetPlatformVal);
+                } catch (IllegalArgumentException iae) {
+                    throw new IOException(
+                            taskHelper.getMessage("err.unknown.target.platform", targetPlatformVal));
+                }
+            }
+        }
+
+        // returns true if the current platform's "jmods" directory is the parent of the
+        // passed javaBasePath
+        private static boolean isJavaBaseFromCurrentPlatform(Path javaBasePath) throws IOException {
```

Comment:
This method checks if we are jlinking the JMOD files from the default module path.   This is not exactly checking if `java.base` matches the current platform since there are other possible setups too.    Suggest to rename the method name to `isJavaBaseFromDefaultModulePath`.   So the comments should also be updated to reflect that.

Reply:
Renamed as suggested and updated the comments as well in the updated version of the PR.